src/backend/distributed/executor/multi_utility.c

```diff
@@ -3173,8 +3187,24 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 
 			if (shouldSyncMetadata)
 			{
-				List *commandList = list_make2(DISABLE_DDL_PROPAGATION,
-											   (char *) ddlJob->commandString);
+				char *setSearchPathCommand = SetSearchPathToCurrentSearchPathCommand();
+				List *commandList = NIL;
+
+				/*
+				 * Given that we're relaying the query to the worker nodes directly,
+				 * we should set the search path exactly the same when necessary.
+				 */
+				if (setSearchPathCommand != NULL)
+				{
+					commandList = list_make3(DISABLE_DDL_PROPAGATION,
```

Comment:
I would have preferred
```
list_make1()
if (setSearchPathCommand)
{
    lappend()
}
lappend()
```

for better readability.

Reply:
makes sense.